src/cmd/cmd_utils.ml

```diff
@@ -34,63 +34,79 @@ let write_testcase =
     in
     res
 
-let add_main_as_start (m : Binary.modul) =
+let find_exported_name exported_names (m : Binary.modul) =
+  List.find_opt
+    (function
+      | { Binary.name; _ } when List.mem name exported_names -> true
+      | _ -> false )
+    m.exports.func
+
+let set_entry_point entry_point (m : Binary.modul) =
   (* We are checking if there's a start function *)
   if Option.is_some m.start then Ok m
   else
     (* If there is none, we look for a function exported with the name `main` *)
```

Comment:
This comment should be updated (and the one below "We found a main function" too)